Mage/src/main/java/mage/abilities/effects/common/combat/CantBeBlockedByAllTargetEffect.java

```diff
@@ -51,4 +51,21 @@ public String getText(Mode mode) {
                 + (filterBlockedBy.getMessage().startsWith("except by") ? "" : "by ")
                 + filterBlockedBy.getMessage();
     }
+
+    @Override
+    public boolean hasHint() {
+        return true;
+    }
+
+    @Override
+    public String getHint(Permanent permanent, Ability source, Game game) {
+        if (!applies(permanent, source, game)) {
```

Comment:
It’s not work. Applies used for basic objects filter. Real check in canAttack, canBlock and other methods. Also it uses different methods with different params and different moment of time to apply.

![IMG_0229](https://github.com/magefree/mage/assets/8344157/3e862f6f-3dae-44e2-921b-713c5904ad82)

Reply:
All I want is a basic object filter - `getHint` needs to just filter which permanent it applies to.